### PR TITLE
Adding empty value for nodeSelector

### PR DESCRIPTION
**Pull request by shubhamcoc**
This will allow to set default value as empty in the values.yaml and close https://github.com/arttor/helmify/issues/106.

### CODE REVIEW

**Comment by shubhamcoc** [comment]
Hi @arttor, Can you review it once? Thanks!!

File: pkg/processor/pod/pod.go [hunk 1]
```diff
@@ -73,14 +74,23 @@ func ProcessSpec(objName string, appMeta helmify.AppMetadata, spec corev1.PodSpe
 	}
 
 	// process nodeSelector if presented:
-	if len(spec.NodeSelector) != 0 {
+	if spec.NodeSelector == nil {
+		return specMap, values, nil
+	} else {
```
Comment (arttor) [inline pkg/processor/pod/pod.go:79]: `else` is redundant if we do return from if branch

File: pkg/processor/pod/pod.go [hunk 2]
```diff
@@ -73,14 +74,23 @@ func ProcessSpec(objName string, appMeta helmify.AppMetadata, spec corev1.PodSpe
...
 		err = unstructured.SetNestedField(specMap, fmt.Sprintf(`{{- toYaml .Values.%s.nodeSelector | nindent 8 }}`, objName), "nodeSelector")
 		if err != nil {
 			return nil, nil, err
 		}
-		err = unstructured.SetNestedStringMap(values, spec.NodeSelector, objName, "nodeSelector")
-		if err != nil {
-			return nil, nil, err
+		if len(spec.NodeSelector) != 0 {
```
Comment (arttor) [inline pkg/processor/pod/pod.go:84]: map len check is effectively equal to nil check - an uninitialized map will return zero len.

**Review by arttor — CHANGES_REQUESTED** [review]
1. is it a good idea to template fields not presented in original manifests? Will it be better to add nodeSelector to source manifests before using helmify?
2. please regenerate chart examples with `cat test_data/sample-app.yaml | go run ./cmd/helmify examples/app` and `cat test_data/k8s-operator-kustomize.output | go run ./cmd/helmify examples/operator` and push changes to your branch

**Comment by shubhamcoc** [comment]
Hi @arttor, the field is present in the source manifest if the value is empty (i.e. nodeSelector: {}), the map len will be zero, as a result the generated manifest will omit the nodeSelector field. Here the check is, if the map is not nil, even if the number of elements can be zero in the map, but it will support default values as empty.

File: pkg/processor/pod/pod.go (hunk 1 above)
Comment (shubhamcoc) [reply pkg/processor/pod/pod.go]: removed it.

File: pkg/processor/pod/pod.go (hunk 2 above)
Comment (shubhamcoc) [reply pkg/processor/pod/pod.go]: removed it.

**Pull request merged**